src/machines/workflow_machines.rs

```diff
@@ -547,29 +555,24 @@ impl WorkflowMachines {
                     self.current_wf_task_commands.push_back(timer);
                 }
                 WFCommand::CancelTimer(attrs) => {
-                    let mkey = *self
-                        .id_to_machine
-                        .get(&CommandID::Timer(attrs.timer_id.to_owned()))
-                        .ok_or_else(|| {
-                            WFMachinesError::MissingAssociatedMachine(format!(
-                                "Missing associated machine for cancelling timer {}",
-                                &attrs.timer_id
-                            ))
-                        })?;
-                    let res = self.machine_mut(mkey).cancel()?;
-                    match res {
-                        MachineResponse::IssueNewCommand(c) => {
-                            self.current_wf_task_commands.push_back(CommandAndMachine {
-                                command: c,
-                                machine: mkey,
-                            })
-                        }
-                        MachineResponse::NoOp => {}
-                        v => {
-                            return Err(WFMachinesError::UnexpectedMachineResponse(
-                                v,
-                                "When cancelling timer",
-                            ));
+                    let m_key =
+                        self.get_machine_key(&CommandID::Timer(attrs.timer_id.to_owned()))?;
+                    let res = self.machine_mut(m_key).cancel()?;
+                    // TODO: Dedupe with req cancel activity
```

Comment:
Let's do this TODO

Reply:
The only concern is PushWFJob arm that exists only in the activity case. Probably there is no harm in having it for both. Let me make a change and we can discuss.